pkg/azuredevops/pipelines/pipelines.go

```diff
@@ -110,6 +133,278 @@ func GetRunLogs(ctx context.Context, buildClient BuildClient, httpClient HTTPCli
+//
+// Parameters:
+// ctx          - The context to control the execution and cancellation of the test.
+// buildClient  - The client interface to interact with the build system.
+// pipelineName - The name of the pipeline within the project.
+// buildID      - A pointer to an integer storing the build identifier.
+// test         - The TimelineTest struct containing the name, expected result, and state of the test stage to be checked.
+//
+// Returns a boolean and an error. The boolean is true if a record matching the test's criteria (name, result, and state)
+// is found in the build timeline. If no matching record is found or if there is an error in retrieving the build timeline,
+// the function returns false and an error with a detailed message for troubleshooting.
+//
+// This function is particularly useful for verifying specific stages or conditions in a build process, especially in continuous
+// integration and deployment scenarios where automated verification of build stages is required.
+func GetBuildStageStatus(ctx context.Context, buildClient BuildClient, projectName string, buildID *int, test TimelineTest) (bool, error) {
+	buildArgs := build.GetBuildTimelineArgs{
+		Project: &projectName,
+		BuildId: buildID,
+	}
+
+	buildTimeline, err := buildClient.GetBuildTimeline(ctx, buildArgs)
+	if err != nil {
+		return false, fmt.Errorf("error getting build timeline: %w", err)
+	}
+
+	return CheckBuildRecords(buildTimeline, test.Name, test.Result, test.State)
+}
+
+// CheckSpecificBuildForCommand searches for the occurrence of a specified command or message within the build logs of a specific build.
+// This function is used to verify if a particular command or log entry was executed or generated during the build process.
+//
+// Parameters:
+// ctx          - The context to control the execution and cancellation of the test.
+// buildClient  - The client interface to interact with the build system.
+// buildID      - A pointer to an integer storing the build identifier.
+// projectName  - The name of the project in which the test is being run.
+// pipelineName - The name of the pipeline within the project.
+// logFinding   - The specific command or message to search for in the build logs.
+// pipelineID   - The identifier of the pipeline.
+//
+// Returns a boolean and an error. The boolean is true if the specified command or message is found in the build logs,
+// indicating that the command was executed. It returns false if the message is not found or if there are no logs for the build.
+// In case of an error in fetching builds or logs, or any other operational issue, the function returns the error with a detailed
+// message for troubleshooting.
+func CheckSpecificBuildForCommand(ctx context.Context, buildClient BuildClient, projectName, pipelineName, logFinding string, pipelineID int, buildID *int) (bool, error) {
```

Comment:
This function and following one `CheckSpecificBuildForMissingCommand` are almost identical except one return statement. Think about merging them in to one and checking if test expect absent or not inside merged function.
This would be equivalent to this test from `RunBuildtests` function.
```if test.ExpectAbsent```

Reply:
Done.